framework/include/vectorpostprocessors/EigenvaluePostprocessor.h

```diff
@@ -0,0 +1,41 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef EIGENVALUEPOSTPROCESSOR_H
+#define EIGENVALUEPOSTPROCESSOR_H
+
+#include "GeneralVectorPostprocessor.h"
+#include "NonlinearEigenSystem.h"
+
+//Forward Declarations
+class EigenvaluePostprocessor;
+
+template<>
+InputParameters validParams<EigenvaluePostprocessor>();
+
+class EigenvaluePostprocessor : public GeneralVectorPostprocessor
```

Comment:
I think you should either add the word "Vector" to this object name or remove Postprocessor. The reason is that we have "Postprocessors" and they mean scalars so this name is a little misleading.